### DISCUSSION TITLE
Shelly 1PM MQTT - disconessione dopo riavvio HA (risolto)

### DISCUSSION THREAD

**Discussion by reiner348** [indomus/forum#2151] [Hardware & Firmware]
Ciao a tutti,
ho un problema con gli Shelly... Ho integrato due Shelly 1PM tramite mqtt seguendo la guida indomus e funzionano perfettamente. L'unico problema che ho è che quando riavvio HA, per qualunque motivo (modifiche, va via la luce, etc.) lo shelly diventa non disponibile. Riavviando anche lui, dopo che HA è ripartito, torna a essere raggiungibile da HA.

Non è un problema  grave, però è fastidioso... E sono gli unici dispositivi MQTT che si comportano così, tutti gli altri (tasmota e switchbot) funzionano perfettamente e ricompaiono da soli anche dopo i riavvi di HA. Qualche idea su come mai gli Shelly mi danno questo problema?

**Comment by snakuzzo** [comment]
Sicuro di aver configurato correttamente l'availability topic?

**Reply by reiner348** [reply]
Credo di sì, anche perchè ho fatto brutale copia-incolla dalla guida, ho solo cambiato il nome dello Shelly per fare in modo che fosse uguale al mio custom MQTT prefix (shelly_SC1)
```
  - name: "shelly_n1"
    state_topic: "shellies/shelly_SC1/relay/0/power"
    availability_topic: "shellies/shelly_SC1/online"
    payload_available: "true"
    payload_not_available: "false"
    unit_of_measurement: "Watt"
```

**Reply by snakuzzo** [reply]
Controlla bene cosa viene pubblicato sul broker con mqtt Explorer o con i mosquitto client 

https://indomus.it/formazione/shelly-mqtt-e-http-comandi-utili/

**Reply by reiner348** [reply]
Ok ci provo. Comunque ora che mi ci hai fatto pensare, lo shelly è l'unico che ha la riga del availability_topic, gli switchbot non ce l'hanno mentre i device tasmota compaiono in automatico tramite l'integrazione tasmota, non li ho proprio configurati nel configuration.yaml (e sottofiles)... Studio un po'!

**Comment by reiner348** [comment]
Non ho capito come fare a vedere cosa viene pubblicato sul broker, purtroppo è un po' lungo da studiare e non riesco in questi giorni. Però ho provato a commentare nel codice la riga dell'availability_topic, così che fosse disattivata, e ora funziona.
Mi viene da pensare che lo Shelly mandi quel messaggio solo quando si avvia / riavvia, poi basta, quindi al riavvio del server HA non lo vede e lo considera non disponibile

**Reply by snakuzzo** [reply]
sì, lo manda quando si avvia e quando viene scatenato da altra publish. Ti consiglio di rimettere quella riga perché se per caso il device va offline non ti accorgerai mai che è offline.
E poi dovresti aggiungere questa automazione per fare in modo che allo start di HA venga scatenata la publish di quel topic...

```yaml
automation:

- alias: MQTT Shelly update
  trigger:
  - event: start
    platform: homeassistant
  action:
  - service: mqtt.publish
    data:
      payload: update
      topic: shellies/command
```

**Reply by reiner348** [reply]
Ok, chiaro, grazie del consiglio! E della automazione già scritta da copiare :)
